### ISSUE TITLE
Update `@yamada-ui/button` accessibility

### ISSUE THREAD

**Issue by setodeve** [yamada-ui/yamada-ui#1737]
### Description
This issue aims to address accessibility deficiencies in our `button component` as per W3C guidelines. 
The component currently does not meet the accessibility standards outlined in the W3C documentation. 

The goal is to ensure the component is accessible to all users, including those using assistive technologies.
Ensure that all unchecked items in this checklist should be addressed.

#### General Requirements
- [x] **Button Role:**
  - [x] The button element has `role="button"`.

- [x] **Accessible Label:**
  - [x] The button has an accessible label computed from its text content, `aria-labelledby`, or `aria-label`.

- [x] **Button Description:**
  - [x] If a description of the button's function is present, the button element has `aria-describedby` set to the ID of the element containing the description.

- [x] **Disabled State:**
  - [x] When the action associated with a button is unavailable, the button has `aria-disabled="true"`.

#### Toggle Button Specific Requirements
- [x] **Toggle State:**
  - [x] The button has an `aria-pressed` attribute.
  - [x] When toggled on, `aria-pressed` is set to `true`.
  - [x] When toggled off, `aria-pressed` is set to `false`.
  - [x] The label does not change when the state changes.

#### Menu Button Specific Requirements
- [x] **Menu Button Role:**
  - [x] The button has `aria-haspopup` set to either `menu` or `true`.

#### Keyboard Interaction
- [x] **Focus Behavior:**
  - [x] When the button has focus:
    - [x] `Space` activates the button.
    - [x] `Enter` activates the button.

- [ ] **Post-Activation Focus:**
  - [x] If the button opens a dialog, focus moves inside the dialog.
  - [ ] If the button closes a dialog, focus typically returns to the button that opened the dialog.
  - [x] If the button's action does not dismiss the current context, focus remains on the button.
  - [ ] If the button's action indicates a context change, focus moves to the starting point for that action.
  - [ ] If activated with a shortcut key, focus usually remains in the context from which the shortcut key was activated.

### Problem Statement/Justification

none

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

### References

https://www.w3.org/WAI/ARIA/apg/patterns/button/

**Comment by chida09** [comment]
>Disabled State:
When the action associated with a button is unavailable, the button has aria-disabled="true".

As few other libraries have adopted aria-disabled for buttons, we will close this issue as not requiring further action.
https://github.com/yamada-ui/yamada-ui/pull/2736

**Comment by chida09** [comment]
>Menu Button Role:
The button has aria-haspopup set to either menu or true.

Already supported.
https://github.com/yamada-ui/yamada-ui/blob/main/packages/components/menu/src/menu-button.tsx#L68

**Comment by chida09** [comment]
>Toggle State:
 The button has an aria-pressed attribute.
 When toggled on, aria-pressed is set to true.
 When toggled off, aria-pressed is set to false.
 The label does not change when the state changes.

I tried it with storybooks and had no problem.

**Comment by chida09** [comment]
>Post-Activation Focus:
If the button closes a dialog, focus typically returns to the button that opened the dialog.

This is going to need to be addressed.

ref:
 - https://mui.com/material-ui/react-dialog/
 - https://react-spectrum.adobe.com/react-spectrum/Dialog.html#dialog

**Comment by setodeve** [comment]
> Accessible Label:
> The button has an accessible label computed from its text content, aria-labelledby, or aria-label.

This component assumes that text will be displayed, so there is no need to set `aria-labelledby` or `aria-label`.

ref:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby

**Comment by taiseiiiii** [comment]
Remaining checkpoint is not necessary in button component.
It is the responsibility of the Dialog component.
So, I am closing this issue because there are few things to address.
